refactor: add menubar metadata to theme editor

sissbruecker opened the pull request:
## Description

Adds metadata for the MenuBar component. 

Also:
- Improves theme detection to work with more complex components. To detect styles for components that have children or overlays we need an instance that has all of these set up. Relying on the selected element seems brittle, as that might not have all of the elements we want to measure, and its custom styles would lead to incorrect results for detecting styles for the global scope. It gets even more difficult for data-driven components like ComboBox. Having a generic setup for all types of components seems impossible, so for now, for each complex component the metadata defines functions on how to properly set up or clean up an instance from which styles can be detected.
- Adds a property editor with a checkbox to allow toggling bold / italic text styles. Could be improved in the future to use a single editor for multiple properties. Checkbox is not styled at the moment.
- Adds a customization to the server rule generation, so that when setting a `border-width` larger than zero, it also sets the `border-style` property to solid. This avoids the user having to set the border style separately for a border to show up.

The generated style rules for the menu bar overlay depend on the overlay class name being set on the menu bar, which means the backend also needs to be changed so that it applies the local class name as overlay class name. In the meanwhile for testing this, the overlay class name can be applied manually:
```java
menuBar.setOverlayClassName("local-class-name");
```

~~PR depends on the changes from https://github.com/vaadin/flow/pull/16372, and is currently set to merge against it.~~

MarcinVaadin reviewed (approved):
It is bit confusing that picking `menu-bar-item` does nothing. Picking `menu-bar` works fine. IMO if child components are included within parent component they shouldn't be picked individually or "redirected" to styling parent component.

IMO border width could also be slider by default.

Bold and italic are fine but eventually should look like in text editors - as button group (maybe with underline also).

PR does LGTM although needs to be addressed in next iteration. Please merge if no further updates in given PR.

jouni commented on vaadin-dev-server/frontend/theme-editor/components/editors/checkbox-property-editor.test.ts:
```diff
@@ -0,0 +1,75 @@
+import './checkbox-property-editor';
+import sinon from 'sinon';
+
+const fontWeightMetadata: CssPropertyMetadata = {
+  propertyName: 'font-weight',
+  displayName: 'Bold',
+  checkedValue: 'bold'
+};
```
Font weight is actually nine different values (100,200,...,800,900), not a boolean. That said, this isn't a big deal at this point, and can be improved later. There might eventually be Lumo design tokens for font weight as well.

sissbruecker replied:
Discussed this, for now a simple toggle is enough.

jouni commented on vaadin-dev-server/frontend/theme-editor/metadata/components/vaadin-menu-bar.ts:
```diff
@@ -0,0 +1,166 @@
+          editorType: EditorType.checkbox,
+          checkedValue: 'bold'
+        },
+        {
+          propertyName: 'font-style',
+          displayName: 'Italic',
+          editorType: EditorType.checkbox,
+          checkedValue: 'italic'
+        }
```
While this is easy to add, I think it adds noise to the editor 99% of the time, as italics are rarely used in UI design.

sissbruecker replied:
OK, let's remove it then. It seemed like something obvious for text, but then again you usually use this in some form of rich text rather than UIs.

jouni commented on vaadin-dev-server/frontend/theme-editor/metadata/components/vaadin-menu-bar.ts:
```diff
@@ -0,0 +1,166 @@
+          displayName: 'Background color',
+          editorType: EditorType.color
+        },
+        {
+          propertyName: 'border-color',
+          displayName: 'Border color',
+          editorType: EditorType.color
+        },
+        {
+          propertyName: 'border-width',
+          displayName: 'Border width'
+        },
```
Missing border-radius (as discussed, all three would always be added together).

sissbruecker commented:
After discussing with Jouni, added a few tweaks:
- Use range slider for border-width
- Added border-radius property
- Removed font-style / italics property

MarcinVaadin reviewed (approved):
Looks better after updates.
